### PR TITLE
Dialogue Line

### PR DESCRIPTION
**Pull request by kefsaj**
Implemented Dialogue Line class
Wrote unit tests for Dialogue Line Class
Added Dialogue Line and Dialogue Line Unit Tests to Makefile

### CODE REVIEW

File: tests/DialogueLineTests.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+DialogueLineTests::DialogueLineTests(const std::string& n): Testable(n) {}
+
+void DialogueLineTests::build()
+{
+    tests.push([](void) -> TestResult {
+        DialogueLine D1("Hello! ", Person());
+        std::string expected = "Hello! ";
+        std::string result = D1.getLineToSay();
+        bool passed = ( expected == result );
```
Comment (BryanMorfe) [inline tests/DialogueLineTests.cpp]: To have this match the style of everything else, remove the additional spaces, result in `bool passed = (expected == result);`

File: tests/DialogueLineTests.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        
+        return TestResult(expected, result, passed);
+    });
+    
+    tests.push([](void) -> TestResult {
+        DialogueLine D1("", Person("11123", "", "", Date::now(), true));
+        std::string expected = "11123";
+        std::string result = D1.getPerson().getIDNum();
+        bool passed = ( expected == result );
```
Comment (BryanMorfe) [inline tests/DialogueLineTests.cpp]: Same as above, remove extra spaces.

File: tests/DialogueLineTests.cpp (hunk 1 above)
Comment (BryanMorfe) [inline tests/DialogueLineTests.cpp]: To have this match the style of everything else, remove the additional spaces, result in `bool passed = (expected == result);`

**Review by BryanMorfe — CHANGES_REQUESTED** [review]
Make the two small changes -- commit, and then I will approve.

File: tests/DialogueLineTests.cpp (hunk 1 above)
Comment (kefsaj) [reply tests/DialogueLineTests.cpp]: Thank you for your insightful comments

**Review by BryanMorfe — APPROVED** [review]

**Pull request merged**